### PR TITLE
[DOC] - Update documentation

### PR DESCRIPTION
**Pull request by TomDonoghue**
Responds to #228 & #226 

It adds a new "Visualizers" section to the docsite. 

It adds a new "Motivations" piece exploring the notion of oscillations as peaks of power in the frequency domain (and some limitations in this). 

It add new examples:
- exporting model parameters to datafames
- an overview / exploration of topics & limitations related to aperiodic parameters (for the current models)
- an overview / exploration of topics & limitations related to periodic parameters (for the current models)

It also updates the Reference page, and adds an in code example of the reporting guidelines, which was added as a new section of the tutorial (I wasn't so clear where to put this, this seemed like a reasonable spot - but it could move). 

Note that across these updates / new materials, these pieces are somewhat drafts that could be extended if people want to add more into them (eg - someone who knows more might want to add more notes to the knee related notes). The goal of this PR is to get the key pieces in, with good first versions, and then further updates can be made as needed as we move into 2.0. 

Note: when this is ready and merged, it should also be merged to `name` (#205), with updates to move to specparam, so that these examples are also properly added to be 2.0 version. 

For ease of checking / reviewing - here is a copy of the built documentation including what's added:
[html.zip](https://github.com/fooof-tools/fooof/files/12076006/html.zip)
^If you download and unzip this, it's a copy of the built documentation page. Open the html folder, and double click 'index.html' to open the home page in a browser. Then you can click on the examples tab, and under the analyses section, click and see the new "Developmental Data Demo" example.

### CODE REVIEW

File: doc/reference.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ In addition, we recommend that reports should include information on:
 Reporting Template & Example
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To assist in reporting on using FOOOF, we have created some templates for reporting on spectral parameterization methods. There are also some utilities included in the code to collected the required information. 
+To assist in reporting on using FOOOF, we have created some templates for reporting on spectral parameterization methods. There are also some utilities included in the code to collected the required information.
```
Comment (ryanhammonds) [inline doc/reference.rst:57]: to collected -> to collect

File: examples/models/plot_aperiodic_params.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+#
+# What we see above matches the common finding that that the offset and exponent are
+# often highly correlated! This is because if you imagine a change in exponent as
+# the spectrum 'rotating' around some frequency value, then (almost) any change in exponent
+# has a corresponding change in offset value! If you note in the above, we actually specified
+# a rotation point around which the exponent changes.
+#
+# This can also be seen in this animation showing this effect across different rotation points:
+#
+# ![spectralrotation](https://raw.githubusercontent.com/fooof-tools/Visualizers/main/gifs/specrot.gif "specrot")
```
Comment (ryanhammonds) [inline examples/models/plot_aperiodic_params.py:107]: When I built the site, this doesn't get rendered.

File: examples/models/plot_aperiodic_params.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+###################################################################################################
+
+# Check the measured aperiodic parameters
+fm.aperiodic_params_
+
+###################################################################################################
+# Knee Frequency
+# ~~~~~~~~~~~~~~
+#
+# You might notice that the knee _parameter_ is not an obvious value. Notably, this parameter
```
Comment (ryanhammonds) [inline examples/models/plot_aperiodic_params.py:147]: \_parameter\_ gets rendered as a literal for me, not italics

File: examples/models/plot_aperiodic_params.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+# value as extracted from the model is something of an abstract quantify based on the
+# formalization of the underlying fit function. A more intuitive measure that we may
+# be interested in is the 'knee frequency', which is an estimate of the frequency value
+# at which the knee occurs.
+#
+# The `:func:`~.compute_knee_frequency` function can be used to compute the knee frequency.
```
Comment (ryanhammonds) [inline examples/models/plot_aperiodic_params.py:153]: I think the first backtick needs to be remove in `:func:. I think it's preventing the linking from happening

File: tutorials/plot_09-Reporting.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+from fooof import __version__ as fooof_version
+print('Current fooof version:', fooof_version)
+
+###################################################################################################
+# Getting Model Reporting Information
+# -----------------------------------
+#
+# To assist with reporting information, the module has the following utilities:
+# - the :func:`~.methods_report_info`, which prints out methods reporting information
+# - the :func:`~.methods_report_text`, which prints out a template of methods reporting text
```
Comment (ryanhammonds) [inline tutorials/plot_09-Reporting.py:61]: the dashes don't get built as bullet points. not sure if that was intended or not

**Review by ryanhammonds — COMMENTED** [review]
Looks good! I left minor comments about rendering. Also, when running `make html`, the build froze at plot_mne_example.py. If ^C'ed the process, it completed the build.

File: examples/models/plot_aperiodic_params.py (hunk 4 above)
Comment (TomDonoghue) [reply examples/models/plot_aperiodic_params.py:153]: Ooops, yeh - good catch! Fixed

File: tutorials/plot_09-Reporting.py (hunk 5 above)
Comment (TomDonoghue) [reply tutorials/plot_09-Reporting.py:61]: I always forget the blank line before a bulleted list! Fixed!

File: examples/models/plot_aperiodic_params.py (hunk 2 above)
Comment (TomDonoghue) [reply examples/models/plot_aperiodic_params.py:107]: Ooops, I did markdown instead of rst. Fixed!

**Comment by TomDonoghue** [comment]
Thanks for the review @ryanhammonds! I fixed the stuff in the comments. 

In terms of the MNE example... hmmm, that seems to be building fine on my side, so I'm not sure what's up. There is a new minimum requirement for MNE after the updates in #259, but if that wasn't up to date I would expect that specific example to fail rather than having it hang. The MNE example is pretty slow - so I wonder if it just hadn't finished yet?

**Pull request merged**